src/test/scala/firrtlTests/ZeroWidthTests.scala

```diff
@@ -315,7 +303,22 @@ class ZeroWidthTests extends FirrtlFlatSpec {
          |    input rwMask: UInt<1>
          |
          |${Seq.tabulate(17)(_ => "    skip").mkString("\n")}""".stripMargin
-    parse(exec(input)) should be(parse(check))
+    compile(input).circuit.serialize should be(parse(check).serialize)
+  }
+
+  "zero width literals" should "be permissible" in {
+    val input =
+      """circuit Foo:
+        |  module Foo:
+        |    output x : UInt<1>
+        |    output y : SInt<3>
+        |
+        |    x <= UInt<0>(0)
+        |    y <= SInt<0>(0)
+        |""".stripMargin
+
+    val result = compile(input).circuit
+    println(result.serialize)
```

Comment:
Can you make this actually check rather than just printing?

Reply:
done